org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/hcr/ScalaHotCodeReplaceManager.scala

```diff
@@ -150,8 +152,17 @@ private[internal] trait HotCodeReplaceExecutor extends Publisher[HCRResult] with
     }
   }
 
+  protected val doHotCodeReplaceBarrier = new AtomicBoolean
+  private def waitForBarrier(): Unit = {
+    if (doHotCodeReplaceBarrier.getAndSet(true)) {
+      while (doHotCodeReplaceBarrier.get) {}
```

Comment:
What do you expect, how long would this loop take? Is the whole race condition happening in < 1ms or in something that is way larger?